fastcrypto-zkp/src/api.rs

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+use std::io::Cursor;
+
+use ark_bls12_381::Bls12_381;
+use ark_groth16::{Proof, VerifyingKey};
+use ark_relations::r1cs::SynthesisError;
+use ark_serialize::CanonicalDeserialize;
+
+use crate::conversions::BlsFr;
+use crate::verifier::{process_vk_special, verify_with_processed_vk};
+
+/// Verify Groth16 proof using the serialized representation of verifying key, input and proof.
+/// This assumes canoncical ser/de defined in Arkworks, instead of the Zcash bytes representation.
+pub fn verify_groth16_in_bytes(
+    vk_bytes: Vec<u8>,
+    x_bytes: Vec<u8>,
+    proof_bytes: Vec<u8>,
```

Comment:
I am not super confident of the separation of the inputs from the three proof points. At least I think they should be renamed as "public inputs" (or similar evocative naming). It also makes sense to me to bundle the three proof points and the scalars together.

Reply:
leaving them separate is much easier for deserialization, what do we gain by bundling them together?

Reply:
I think @kchalkias expressed my thinking better here: https://github.com/MystenLabs/fastcrypto/pull/223/files#r1023359491 TL;DR, we should make clear, through renaming variables, that both the proof points and the public inputs are part of a proof instance.
One example : `proof_points_as_bytes`, `proof_public_inputs_as_bytes`